parts/linux/cloud-init/artifacts/cse_config.sh

```diff
@@ -560,6 +568,13 @@ configGPUDrivers() {
     if [[ $OS == $UBUNTU_OS_NAME ]]; then
         retrycmd_if_failure 120 5 25 nvidia-modprobe -u -c0 || exit $ERR_GPU_DRIVERS_START_FAIL
     fi
+
+    # Noticed that NVIDIA driver version R515 and later versions will experience long loading time of nvidia-smi on 
+    # ND A100-v4 and NDm A100-v4 series. For now, use the workaround solution to enable NVIDIA persistence mode which 
+    # will accelerate the loading time of future nvidia-smi operations
+    if [[ $OS == $MARINER_OS_NAME ]]; then
+        retrycmd_if_failure 120 5 35 nvidia-smi -pm 1 || exit $ERR_GPU_DRIVERS_START_FAIL
```

Comment:
I was also able to capture this on Ubuntu when testing using ab e2e. One thing I would like to open for discussion is that "nvidia-smi -pm 1" will also take quite long to complete (> 25sec on Mariner), in that case, from the perspective of total cluster creation time, do we go for enabling persistence mode and then check "nvidia-smi" or simply increasing the timeout of nvidia-smi?

Reply:
when do we get the benefits of persistence mode? on reboot when we don't have to reload nvidia-smi? or any time we invoke it?

Reply:
After enabling persistence mode, we'll expect nvidia-smi or any applications that attempt to trigger GPU initialization to complete much faster. But this will only occur in the current session. If you do a reboot, then persistence mode will be disabled and needs to re-activated. 

Turning on persistence mode will also bring the benefits of data persistence. Otherwise the driver state won't be preserved and it will revert back to default when the GPU is initialized again next time.

Reply:
Replaced "nvidia-smi -pm 1" with using nvidia persistence daemon instead because "nvidia-smi -pm 1" is a legacy solution that will be deprecated.